sdk/containerregistry/Azure.Containers.ContainerRegistry/src/ContainerRegistryClient.cs

```diff
@@ -56,69 +54,94 @@ protected ContainerRegistryClient()
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         public virtual AsyncPageable<string> GetRepositoriesAsync(CancellationToken cancellationToken = default)
         {
-            return PageResponseEnumerator.CreateAsyncEnumerable(async (continuationToken, pageSizeHint) =>
+            async Task<Page<string>> FirstPageFunc(int? pageSizeHint)
             {
                 using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(ContainerRegistryClient)}.{nameof(GetRepositories)}");
                 scope.Start();
                 try
                 {
-                    Response<Repositories> response =
-                        await _restClient.GetRepositoriesAsync(
-                            continuationToken,
-                            pageSizeHint,
-                            cancellationToken)
-                       .ConfigureAwait(false);
-
-                    string lastRepository = null;
-                    if (!string.IsNullOrEmpty(response.Value.Link))
-                    {
-                        Uri nextLink = new Uri(response.Value.Link);
-                        NameValueCollection queryParams = HttpUtility.ParseQueryString(nextLink.Query);
-                        lastRepository = queryParams["last"];
-                    }
-
-                    return Page<string>.FromValues(response.Value.RepositoriesValue, lastRepository, response.GetRawResponse());
+                    ResponseWithHeaders<Repositories, ContainerRegistryGetRepositoriesHeaders> response = await _restClient.GetRepositoriesAsync(last: null, n: pageSizeHint, cancellationToken).ConfigureAwait(false);
+                    return Page.FromValues(response.Value.RepositoriesValue, response.Headers.Link, response.GetRawResponse());
                 }
-                catch (Exception ex)
+                catch (Exception e)
                 {
-                    scope.Failed(ex);
+                    scope.Failed(e);
                     throw;
                 }
-            });
+            }
+
+            async Task<Page<string>> NextPageFunc(string continuationToken, int? pageSizeHint)
+            {
+                using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(ContainerRegistryClient)}.{nameof(GetRepositories)}");
+                scope.Start();
+                try
+                {
+                    string uriReference = ParseUriReferenceFromLinkHeader(continuationToken);
+                    ResponseWithHeaders<Repositories, ContainerRegistryGetRepositoriesHeaders> response = await _restClient.GetRepositoriesNextPageAsync(uriReference, last: null, n: pageSizeHint, cancellationToken).ConfigureAwait(false);
```

Comment:
Are the `last` and `n` parameters ever used? Seems to not be used here - https://github.com/Azure/azure-sdk-for-net/blob/0da3d56db51aa2906031bd788bad2975fa20a4ae/sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Generated/ContainerRegistryRestClient.cs#L219

Reply:
curious why `n` and `last` are not included in the `nextLink` property? is the expectation that the user will modify the size of each request?

Reply:
`n` and `last` are part of the `nextLink` returned by the service.  It's a quirk of how the code generator handles `x-ms-pageable` that it generated this method this way ...  I have an issue to update the swagger to do it more cleanly that I hope to address in a later sprint: https://github.com/azure/azure-sdk-for-net/issues/19833

Per your comment, @JoshLove-msft , I'll pass null for both `last` and `n` for now so there isn't confusion about whether an input value is being used.  Thanks for catching this!